lte/gateway/c/core/oai/test/s1ap_task/test_s1ap_mme_handlers.cpp

```diff
@@ -1991,6 +2062,185 @@ TEST_F(S1apMmeHandlersTest, HandleS1apErabModificationCnf) {
+                          0x00, 0x20, 0x00, 0x3c, 0x40, 0x0b, 0x80, 0x09, 0x22,
+                          0x52, 0x41, 0x44, 0x49, 0x53, 0x59, 0x53, 0x22, 0x00,
+                          0x40, 0x00, 0x07, 0x00, 0x00, 0x00, 0x40, 0x00, 0xf1,
+                          0x10, 0x00, 0x89, 0x40, 0x01, 0x00};
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          s1_bytes_2, sizeof(s1_bytes_2), state, assoc_id, stream_id),
+      RETURNok);
+
+  // Simulate InitialUEMessage - Attach Request
+  uint8_t initial_ue_bytes[] = {
+      0x00, 0x0c, 0x40, 0x48, 0x00, 0x00, 0x05, 0x00, 0x08, 0x00, 0x02,
+      0x00, 0x01, 0x00, 0x1a, 0x00, 0x20, 0x1f, 0x07, 0x41, 0x71, 0x08,
+      0x09, 0x10, 0x10, 0x00, 0x00, 0x00, 0x00, 0x10, 0x02, 0xe0, 0xe0,
+      0x00, 0x04, 0x02, 0x01, 0xd0, 0x11, 0x40, 0x08, 0x04, 0x02, 0x60,
+      0x04, 0x00, 0x02, 0x1c, 0x00, 0x00, 0x43, 0x00, 0x06, 0x00, 0x00,
+      0xf1, 0x10, 0x00, 0x01, 0x00, 0x64, 0x40, 0x08, 0x00, 0x00, 0xf1,
+      0x10, 0x00, 0x00, 0x00, 0xa0, 0x00, 0x86, 0x40, 0x01, 0x30};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          initial_ue_bytes, sizeof(initial_ue_bytes), state, assoc_id,
+          stream_id),
+      RETURNok);
+
+  handle_mme_ue_id_notification(state, assoc_id);
+
+  // Simulate UECapabilityInfoIndication
+  uint8_t ue_cap_bytes[] = {
+      0x00, 0x16, 0x40, 0x53, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x02,
+      0x00, 0x07, 0x00, 0x08, 0x00, 0x02, 0x00, 0x01, 0x00, 0x4a, 0x40,
+      0x40, 0x3f, 0x01, 0xe8, 0x01, 0x03, 0xac, 0x59, 0x80, 0x07, 0x00,
+      0x08, 0x20, 0x81, 0x83, 0x9b, 0x4e, 0x1c, 0x3f, 0xf8, 0x7f, 0xf0,
+      0xff, 0xe1, 0xff, 0xc3, 0xff, 0x87, 0xff, 0x0f, 0xfe, 0x1f, 0xfd,
+      0xf8, 0x37, 0x62, 0x78, 0x00, 0xa0, 0x18, 0x5f, 0x80, 0x00, 0x00,
+      0x00, 0x1c, 0x07, 0xe0, 0xdd, 0x89, 0xe0, 0x00, 0x00, 0x00, 0x07,
+      0x09, 0xf8, 0x37, 0x62, 0x78, 0x00, 0x00, 0x00, 0x00, 0x00};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          ue_cap_bytes, sizeof(ue_cap_bytes), state, assoc_id, stream_id),
+      RETURNok);
+
+  // Generate downlink nas transport with dummy payload
+  bstring p;
+  std::string test_str = "test";
+  STRING_TO_BSTRING(test_str, p);
+  s1ap_generate_downlink_nas_transport(state, 1, 7, &p, 1, &is_state_same);
+  bdestroy_wrapper(&p);
+
+  // Authentication response proc packet bytes
+  uint8_t auth_bytes[] = {
+      0x00, 0x0d, 0x40, 0x3d, 0x00, 0x00, 0x05, 0x00, 0x00, 0x00, 0x02,
+      0x00, 0x07, 0x00, 0x08, 0x00, 0x02, 0x00, 0x01, 0x00, 0x1a, 0x00,
+      0x14, 0x13, 0x07, 0x53, 0x10, 0x1e, 0x63, 0x7e, 0x5c, 0x58, 0xec,
+      0x5a, 0xa8, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+      0x64, 0x40, 0x08, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x00, 0x00, 0xa0,
+      0x00, 0x43, 0x40, 0x06, 0x00, 0x00, 0xf1, 0x10, 0x00, 0x01};
+
+  ASSERT_EQ(
+      simulate_pdu_s1_message(
+          auth_bytes, sizeof(auth_bytes), state, assoc_id, stream_id),
+      RETURNok);
+
```

Comment:
For the purpose of this test, you may skip Auth Response and Security Mode Complete messages, since they do not alter S1ap state. They are just forwarded to the MME app task.

Reply:
Correct, I skipped them for sake of unit testing S1AP handlers procedures only